DeceptionGame/Assets/Scripts/MapGeneration/MapGenerator.cs

```diff
@@ -0,0 +1,559 @@
+﻿using UnityEngine;
```

Comment:
This is where the "magic" happens. I tried to comment it as much as I can. Conceptually, it is very simple, but there is a reference if you need it.